app/services/record_declarations/base.rb

```diff
@@ -26,6 +26,15 @@ def profile_exists
       end
     end
 
+    def date_has_the_right_format
```

Comment:
are you able to make this private if only used by validation?

Reply:
personally i like to prefix validation methods with `validate_` that way i know it'll add errors potentially. otherwise someone might think `date_has_the_right_format` just returns a boolean with no side effects. i guess this is optional

Reply:
Hmm, so then the top of the file would read
```
validate :validate_xxx
```

Feels a bit awkward. I made it and a few other methods private.